### ISSUE TITLE
Allow better control over rule violations

### ISSUE THREAD

**Issue by mithun** [cdklabs/cdk-nag#402]
Attempting to use `cdk-nag` in an existing codebase can be a daunting task. As a developer, I'd like to have the capability to ease an existing codebase into compliance with the rules by first having `cdk-nag` only provide rule violations as informational messages (as a non-default behavior).

**Comment by dontirun** [comment]
Hi! Thanks for raising this issue. While it's not difficult to add this functionally, I'd like to discuss whether or not it's a good feature to add.

Currently, I don't think converting all errors into warnings or informational is the correct way to go about this. In my experience a critical portion of compliance is about the documention of risks, mitigations, and exceptions. Using `cdk-nag` in a warning only mode won't prevent applications in pipelines and command line deployments from launching. It may also give a false sense of compliance seeing that `cdk-nag` is included in the code  and the application has deployed.

My current thoughts here are to use the [suppression system](https://github.com/cdklabs/cdk-nag#suppressing-a-rule) to document that the issues will be addressed in the future.

That being said I do agree that in a large code base this can be tedious. My thoughts are to perhaps add a global suppression. That way developers have easier time with integration while documenting the in progress nature.

Would love to hear thoughts on this!

**Comment by cmilbert** [comment]
I agree using the rule suppression system is a better way to address the issues when bolting on cdk-nag to existing large code bases.  By adding an ignore rule there is a conscious decision being applied against a failed rule.  Having a flag to drop an error down to a warning potentially creates a false sense of validation.

Another beneficial feature would be to have a list of suppressed rules shown as warnings at the end of validation.  I will create a new issue for this feature.
